### PR TITLE
gcp metadata: refactor and fix confusing logs

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/43185 by removing the log to
debug level; it previously was not logged at all.

Additionally, refactored to make this substantially simpler.

This should have no user facing changes beyond the logging being
removed.

### CODE REVIEW

File: pkg/bootstrap/platform/gcp.go [hunk 1]
```diff
@@ -156,64 +178,28 @@ func (e *gcpEnv) shouldFillMetadata() bool {
...
-	if !e.shouldFillMetadata() {
-		e.metadata = md
-		return md
-	}
+	// Start off with static metadata (if there is any)
+	md := maps.Clone(GCPStaticMetadata)
 
+	// suppliers is an array of functions that supply the metadata for missing properties
 	suppliers := []metadataSupplier{
+		createMetadataSupplier(GCPProject, projectIDFn),
```
Comment (aryan16) [inline pkg/bootstrap/platform/gcp.go:199]: we should check these fields first from GCP_METADATA and if it is not available then we should rely on compute metadata server ?

Comment (howardjohn) [reply pkg/bootstrap/platform/gcp.go:199]: Well we do get them from GCP_METADATA. but then possible override them. yeah I guess we should probably prefer the env, will update it. good idea

Comment (howardjohn) [reply pkg/bootstrap/platform/gcp.go:199]: This is already handled actually, I forgot: https://github.com/istio/istio/blob/ac876ff08c16859048e3b2d959ca9c09e8de3b5a/pkg/bootstrap/platform/gcp.go#L226

**Review by aryan16 — APPROVED** [review]

**Pull request merged**